### ISSUE TITLE
support precise types from `enum.property` getter

### ISSUE THREAD

**Issue by carljm** [astral-sh/ty#3681]
```py
from enum import Enum, property as enum_property
from typing import assert_type

class Animal(Enum):
    LION = 1

    @enum_property
    def sound(self) -> str:
        return "roar"

# Other type checkers pass this assertion, we infer `Any`
assert_type(Animal.LION.sound, str)
```

Typeshed [annotates `enum.property`](https://github.com/python/typeshed/blob/main/stdlib/enum.pyi#L183) as a subclass of `DynamicClassAttribute`, which is [a descriptor](https://github.com/python/typeshed/blob/main/stdlib/types.pyi#L655-L675) inheriting from builtin `property` (this part is a lie to help pyright), but with a `__get__` annotated to return `Any`. Currently we do not treat subclasses of `property` as having all the behaviors of the built-in `property` -- and even if we did, we would see the override of `__get__` returning `Any` and (correctly IMO) consider that as overriding the builtin `property` behavior.

It seems that pyright relies on the "inherit builtin property" (and ignores the `__get__` override), while other type checkers special-case `enum.property` as having the same behavior as built-in `property`. I think we should do the latter.
